tools/clang/tools/dxcompiler/dxcutil.cpp

```diff
@@ -177,17 +149,14 @@ HRESULT ValidateAndAssembleToContainer(AssembleInputs &inputs) {
   std::unique_ptr<llvm::Module> llvmModuleWithDebugInfo;
 
   CComPtr<IDxcValidator> pValidator;
-  bool bInternalValidator = CreateValidator(pValidator, inputs.SelectValidator);
-  // Warning on internal Validator
+  CreateValidator(pValidator);
 
   CComPtr<IDxcValidator2> pValidator2;
-  if (!bInternalValidator) {
-    pValidator.QueryInterface(&pValidator2);
-  }
+  pValidator.QueryInterface(&pValidator2);
 
-  if (bInternalValidator || pValidator2) {
-    // If using the internal validator or external validator supports
-    // IDxcValidator2, we'll use the modules directly. In this case, we'll want
+  if (pValidator2) {
+    // If IDxcValidator2 is supported in the given validator,
```

Comment:
No need for `IDxcValidator2` check, we will always use the internal interface.

I think there are a few more adjustments needed in this function.

For instance, this diagnostic text needs revision, or we need to consider whether this should even be possible at this point:
> The module cannot be validated by the version of the validator currently attached.

Reply:
I thought about it, I think we should leave the comparison in. I could remove it, but leaving it in means we get a diagnostic when validating a module with a higher minimum requirement level than the internal validator (which shouldn't be possible in practice, but serves as a test case for when future compiler versions are used with DXIL IR that's even further in the future. 
I'd think, say, Dxc version 1.10 should say something if it receives a module with version 1.11.

Reply:
The validator will fail with an error if you try to validate a module with a newer validator version than it supports.  We shouldn't need the version check here because we are validating something we produced, which should have a supported validator version, and otherwise, we can still rely on the validator to emit an error.